src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -77,24 +72,200 @@ type persistenceFlush struct {
 	timeRange   xtime.Range
 }
 
+type instrumentation struct {
+	opts                               Options
+	log                                *zap.Logger
+	nowFn                              clock.NowFn
+	bootstrapDataDuration              tally.Timer
+	bootstrapIndexDuration             tally.Timer
+	bootstrapShardsDuration            tally.Timer
+	persistedIndexBlocksOutOfRetention tally.Counter
+	start                              time.Time
+	startShards                        time.Time
+}
+
+func newInstrumentation(opts Options) *instrumentation {
+	instrumentOptions := opts.ResultOptions().InstrumentOptions()
+	scope := instrumentOptions.MetricsScope().SubScope("peers-bootstrapper")
+	instrumentOptions = instrumentOptions.SetMetricsScope(scope)
+	return &instrumentation{
+		opts:                               opts,
+		log:                                instrumentOptions.Logger().With(zap.String("bootstrapper", "peers")),
+		nowFn:                              opts.ResultOptions().ClockOptions().NowFn(),
+		bootstrapDataDuration:              scope.Timer("peer-bootstrap-data-duration"),
+		bootstrapIndexDuration:             scope.Timer("peer-bootstrap-index-duration"),
+		bootstrapShardsDuration:            scope.Timer("peer-bootstrap-shards-duration"),
+		persistedIndexBlocksOutOfRetention: scope.Counter("persist-index-blocks-out-of-retention"),
```

Comment:
I think the `peer-bootstrap-...` prefix for these timers is redundant, since they are under the `peers-bootstrapper` scope (see above). E.g. the persist index block  metric shows up in Grafana with a name `peers_bootstrapper_persist_index_blocks_out_of_retention`. Similarly, data duration metric would be named `peers_bootstrapper_peer_bootstrap_data_duration`.